src/routers/datatypeRouters/edges/genes_regulatory_regions.ts

```diff
@@ -0,0 +1,107 @@
+import { z } from 'zod'
+import { db } from '../../../database'
+import { QUERY_LIMIT } from '../../../constants'
+import { publicProcedure } from '../../../trpc'
+import { loadSchemaConfig } from '../../genericRouters/genericRouters'
+import { paramsFormatType } from '../_helpers'
+import { descriptions } from '../descriptions'
+import { TRPCError } from '@trpc/server'
+import { geneSearch, singleGeneQueryFormat } from '../nodes/genes'
+import { commonNodesParamsFormat } from '../params'
+
+const MAX_PAGE_SIZE = 500
+
+const schema = loadSchemaConfig()
+
+const predictionFromGeneFormat = z.object({
+  gene: z.object({
+    name: z.string(),
+    id: z.string(),
+    start: z.number(),
+    end: z.number(),
+    chr: z.string()
+  }),
+  predictions: z.array(z.object({
+    id: z.string(),
+    cell_type: z.string(),
+    score: z.number(),
+    model: z.string(),
+    dataset: z.string(),
+    enhancer_type: z.string(),
+    enhancer_start: z.number(),
+    enhancer_end: z.number()
+  }))
+})
+
+const regulatoryRegionToGeneSchema = schema['regulatory element to gene expression association']
+
+async function findPredictionsFromGene (input: paramsFormatType): Promise<any> {
+  let limit = QUERY_LIMIT
+  if (input.limit !== undefined) {
+    limit = (input.limit as number <= MAX_PAGE_SIZE) ? input.limit as number : MAX_PAGE_SIZE
+    delete input.limit
+  }
+
+  const page = input.page as number
+
+  input.page = 0
+  input.organism = 'Homo sapiens'
+  const genes = (await geneSearch(input))
+
+  if (genes.length === 0) {
+    throw new TRPCError({
+      code: 'NOT_FOUND',
+      message: 'Gene not found.'
+    })
+  }
+
+  // assuming unique return
+  // input params are IDs only
+  const gene = genes[0]
```

Comment:
Sometime it still possible to get more than one gene based on the gene filters in genesCommonQueryFormat. I think alias may give you more than one gene sometimes.